src/dmd/backend/obj.d

```diff
@@ -845,409 +676,114 @@ version (OMFandMSCOFF)
-        {
-            return Obj_reftodatseg(seg, offset, val, targetdatum, flags);
-        }
-
-        void reftofarseg(int seg, targ_size_t offset, targ_size_t val, int farseg, int flags)
-        {
-        }
-
-        void reftocodeseg(int seg, targ_size_t offset, targ_size_t val)
-        {
-            return Obj_reftocodeseg(seg, offset, val);
-        }
-
-        int reftoident(int seg, targ_size_t offset, Symbol *s, targ_size_t val, int flags)
-        {
-            return Obj_reftoident(seg, offset, s, val, flags);
-        }
-
-        void far16thunk(Symbol *s)
-        {
-            return Obj_far16thunk(s);
-        }
-
-        void fltused()
-        {
-            return Obj_fltused();
-        }
-
-        int data_readonly(char *p, int len, int *pseg)
-        {
-            return Obj_data_readonly(p, len, pseg);
-        }
-
-        int data_readonly(char *p, int len)
-        {
-            return Obj_data_readonly(p, len);
-        }
-
-        int string_literal_segment(uint sz)
-        {
-            return Obj_string_literal_segment(sz);
-        }
-
-        Symbol *sym_cdata(tym_t ty, char *p, int len)
-        {
-            return Obj_sym_cdata(ty, p, len);
-        }
-
-        void func_start(Symbol *sfunc)
-        {
-            return Obj_func_start(sfunc);
-        }
 
-        void func_term(Symbol *sfunc)
-        {
-            return Obj_func_term(sfunc);
-        }
 
-        void write_pointerRef(Symbol* s, uint off)
-        {
-            return Obj_write_pointerRef(s, off);
-        }
-
-        int jmpTableSegment(Symbol* s)
-        {
-            return Obj_jmpTableSegment(s);
-        }
-
-        Symbol *tlv_bootstrap()
-        {
-            return Obj_tlv_bootstrap();
-        }
-
-        void gotref(Symbol *s)
-        {
-            return Obj_gotref(s);
-        }
+extern __gshared Obj objmod;
 
-        uint addstr(Outbuffer *strtab, const(char)* p)
-        {
-            return Obj_addstr(strtab, p);
-        }
+/*****************************************
+ * Use to generate 4 function declarations, one for
+ * each object file format supported.
+ * Params:
+ *      pattern = function declaration
+ * Returns:
+ *      declarations as a string suitable for mixin
+ */
+private extern (D)
+string ObjMemDecl(string pattern)
+{
+    string r =
+        gen(pattern,    "Omf") ~ ";\n" ~
+        gen(pattern, "MsCoff") ~ ";\n" ~
+        gen(pattern,    "Elf") ~ ";\n" ~
+        gen(pattern,   "Mach") ~ ";\n";
+    return r;
+}
 
-        Symbol *getGOTsym()
-        {
-            return Obj_getGOTsym();
-        }
+/****************************************
+ * Generate boilerplate for static dispatch that
+ * returns a void.
+ * Params:
+ *      arg = string to insert where ever there's a '$'
+ * Returns:
+ *      mixin string with static dispatch
+ */
+private extern (D)
+string genRetVoid(string arg)
```

Comment:
Is there need to have a special-case `genRetVoid`? I think you can just use `genRetVal` always since D allows void expressions in the return statement of a void function.

Reply:
It reduces size of generated code.